File: apple/Sources/ReactNativeWatchOS/Modifiers/BackgroundStyle.swift
import SwiftUI

/// SwiftUI `.backgroundStyle(_:)` (watchOS 9+). Sets the default style for
/// the backgrounds of views within this view. Resolves the `style` string
/// through the shared `RNWShapeStyleParser`; leaves the view unchanged when
/// unrecognized.
enum RNWBackgroundStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("backgroundStyle") { view, params, _ in
            guard let style = RNWShapeStyleParser.parse(params.string("style")) else {
                return view
            }
            return AnyView(view.backgroundStyle(style))
        }
    }
}

File: src/modifiers/backgroundStyle.ts
import { createModifier } from './createModifier';

/// SwiftUI `.backgroundStyle(_:)` (watchOS 9+). Sets the default style for
/// the backgrounds of views within this view. The `style` string supports
/// named / `#RRGGBB` / `#RRGGBBAA` colors, SwiftUI semantic colors,
/// hierarchical levels, `'tint'`, and (watchOS 10+) materials — resolved by
/// the shared shape-style parser.
export function backgroundStyle(style: string) {
  return createModifier('backgroundStyle', { style });
}
